servo/runner.py

```diff
@@ -188,6 +193,35 @@ async def exec_command(self) -> servo.api.Status:
         else:
             raise ValueError(f"Unknown command '{cmd_response.command.value}'")
 
+    async def diagnostics_check(self):
+
+        while self._running:
+            try:
+                self.logger.trace("Polling for diagnostics request")
+
+                request = await self._diagnostics_request()
+
+                if request == servo.api.DiagnosticStates.withhold:
+                    self.logger.trace("Withholding diagnostics")
+
+                elif request == servo.api.DiagnosticStates.send:
+                    self.logger.info(f"Diagnostics requested, gathering and sending")
+                    diagnostic_data = await self._get_diagnostics()
+                    send = await self._post_diagnostics(diagnostic_data)
+                    reset = await self._reset_diagnostics()
+
+                elif request == servo.api.DiagnosticStates.stop:
+                    self.logger.info(f"Received request to disable polling for diagnostics")
+                    asyncio.current_task().cancel()
+                else:
+                    raise
+
+                await asyncio.sleep(10)
+
+            except Exception as error:
+                self.logger.exception(f"failed with unrecoverable error: {error}")
+                raise error
+
```

Comment:
Please move to a new class (eg. DiagnosticsChecker) in servo/telemetry.py

Reply:
Ok, I figured it was a part of the API but makes sense to silo. I am having trouble understanding how the Diagnostics class should access the running servo however. If I make a `class DiagnosticsHandler(servo.api.Mixin)` in servo/telemetry.py, what is the correct way for it to operate on the running servo within the runner?

Reply:
You can get the servo for the current context (usually only one context/servo per assembly anyway) by using the `servo.current_servo` method https://github.com/opsani/servox/blob/e5d744cb298f6a93d1c55354ba6300f246be67e8/servo/servo.py#L29

Also note its okay for the runner to continue managing the repeating task (since the runner also [sets up cancellation handling](https://github.com/opsani/servox/blob/e5d744cb298f6a93d1c55354ba6300f246be67e8/servo/runner.py#L393) and needs to be aware of the diagnostics task so its not stopped indefinitely (or at all)). The code run by the task should be part of telemetry.py and the runner is responsible for init (and, optionally, calling a simple start method if it makes sense to implement one outside of the init)